packages/transformer-compile-class/src/index.ts

```diff
@@ -1,12 +1,33 @@
 import type { SourceCodeTransformer } from '@unocss/core'
-import { escapeRegExp, expandVariantGroup } from '@unocss/core'
+import { expandVariantGroup } from '@unocss/core'
 
 export interface CompileClassOptions {
   /**
-   * Trigger string
-   * @default ':uno:'
+   * Trigger regex literal. The default trigger regex literal matches `:uno:`,
+   * for example: `<div class=":uno: font-bold text-white">`.
+   *
+   * @example
+   * The trigger additionally allows defining a capture group named `name`, which
+   * allows custom class names. One possible regex would be:
+   *
+   * ```
+   * export default defineConfig({
+   *   transformers: [
+   *     transformerCompileClass({
+   *       trigger: /(["'`]):uno(?:-)?(?<name>[^\s\1]+)?:\s([^\1]*?)\1/g
+   *     }),
+   *   ],
+   * })
+   * ```
+   *
+   * This regular expression matches `:uno-MYNAME:` and uses `MYNAME` in
+   * combination with the class prefix as the final class name, for example:
+   * `.uno-MYNAME`. It should be noted that the regex literal needs to include
+   * the global flag `/g`.
+   *
+   * @default `/(["'`]):uno:\s([^\1]*?)\1/g`
    */
-  trigger?: string
+  trigger?: RegExp
```

Comment:
```suggestion
  trigger?: string | RegExp
```

To keep backward compatibility, we could allow strings that do extract match (without the capability if custom name)

Reply:
Changed in c00c5e5